torch/lib/THD/base/data_channels/DataChannelGloo.cpp

```diff
@@ -69,11 +73,35 @@ DataChannelGloo::DataChannelGloo(InitMethod::Config config)
 {
   _num_processes = config.world_size;
 
+#if defined(WITH_GLOO_IBVERBS) && WITH_GLOO_IBVERBS
+
+  // This helper function automatically detects the IB device in the system
+  auto ibDeviceNames = ::gloo::transport::ibverbs::getDeviceNames();
+
+  if (!ibDeviceNames.size()) {
```

Comment:
nit: `if (ibDeviceNames.empty())` is cleaner

Reply:
Will change